src/Build/BackEnd/Components/RequestBuilder/TargetEntry.cs

```diff
@@ -652,7 +663,7 @@ internal async Task ExecuteTarget(ITaskBuilder taskBuilder, BuildRequestEntry re
                     targetLoggingContext?.LogTargetBatchFinished(projectFullPath, targetSuccess, targetOutputItems?.Count > 0 ? targetOutputItems : null);
                 }
 
-                _targetResult = new TargetResult(targetOutputItems.ToArray(), aggregateResult, targetLoggingContext?.BuildEventContext);
+                _targetResult = new TargetResult(targetOutputItems?.ToArray() ?? Array.Empty<TaskItem>(), aggregateResult, targetLoggingContext?.BuildEventContext);
```

Comment:
I don't think eliminating one list allocation per target will be measurable, but it makes code little better.
I think the code can be safer if we initialize it to `Array.Empty<TaskItem>()` as oppose to null and we don't need to handle null here or future elsewhere.

Reply:
It's a little more complicated because targetOutputItems is a List, and there's no allocation-free variant of that, but I can make targetOutputItems an array with Array.Empty<TaskItem> then initialize a List if we're keeping dupes and not if we're using a HashSet, and that's probably safer.